src/runner/trace.cpp

```diff
@@ -18,13 +20,45 @@ void Trace::UnregisterProvider()
     TraceLoggingUnregister(g_hProvider);
 }
 
-void Trace::EventLaunch(const std::wstring& versionNumber)
+void Trace::EventLaunch(const std::wstring& versionNumber, bool isProcessElevated)
 {
     TraceLoggingWrite(
         g_hProvider,
         "Runner_Launch",
         TraceLoggingWideString(versionNumber.c_str(), "Version"),
         ProjectTelemetryPrivacyDataTag(ProjectTelemetryTag_ProductAndServicePerformance),
         TraceLoggingBoolean(TRUE, "UTCReplace_AppSessionGuid"),
-        TraceLoggingKeyword(PROJECT_KEYWORD_MEASURE));
+        TraceLoggingKeyword(PROJECT_KEYWORD_MEASURE),
+        TraceLoggingBoolean(isProcessElevated, "Elevated"));
+}
+
+void Trace::SettingsChanged(const GeneralSettings& settings)
+{
+    std::wstring enabledModules;
+    for (const auto& [name, isEnabled] : settings.isModulesEnabledMap)
+    {
+        if (isEnabled)
+        {
+            if (!enabledModules.empty())
+            {
+                enabledModules += L", ";
+            }
+
+            enabledModules += name;
+        }
+    }
+
+    TraceLoggingWrite(
+        g_hProvider,
+        "GeneralSettingsChanged",
+        ProjectTelemetryPrivacyDataTag(ProjectTelemetryTag_ProductAndServicePerformance),
+        TraceLoggingBoolean(settings.isPackaged, "Packaged"),
+        TraceLoggingBoolean(settings.isStartupEnabled, "RunAtStartup"),
+        TraceLoggingWideString(settings.startupDisabledReason.c_str(), "StartupDisabledReason"),
+        TraceLoggingWideString(enabledModules.c_str(), "ModulesEnabled"),
+        TraceLoggingBoolean(settings.isElevated, "Elevated"),
+        TraceLoggingBoolean(settings.isRunElevated, "AlwaysRunElevated"),
+        TraceLoggingWideString(settings.theme.c_str(), "Theme"),
+        TraceLoggingWideString(settings.systemTheme.c_str(), "SystemTheme"),
+        TraceLoggingWideString(settings.powerToysVersion.c_str(), "PowerToysVersion"));
```

Comment:
Here, you need to add a few fields, basically it has to look like this:
```
    TraceLoggingWrite(
        g_hProvider,
        "GeneralSettingsChanged",
        TraceLoggingBoolean(settings.isStartupEnabled, "RunAtStartup"),
        TraceLoggingWideString(settings.startupDisabledReason.c_str(), "StartupDisabledReason"),
        TraceLoggingWideString(enabledModules.c_str(), "ModulesEnabled"),
        TraceLoggingBoolean(settings.isRunElevated, "AlwaysRunElevated"),
        TraceLoggingWideString(settings.theme.c_str(), "Theme"),
        ProjectTelemetryPrivacyDataTag(ProjectTelemetryTag_ProductAndServicePerformance),
        TraceLoggingBoolean(TRUE, "UTCReplace_AppSessionGuid"),
        TraceLoggingKeyword(PROJECT_KEYWORD_MEASURE));
```

I've removed 
```
TraceLoggingBoolean(settings.isPackaged, "Packaged"),
TraceLoggingWideString(settings.systemTheme.c_str(), "SystemTheme"),
```
since we don't need them.

I've also removed
```
TraceLoggingBoolean(settings.isElevated, "Elevated"),
TraceLoggingWideString(settings.powerToysVersion.c_str(), "PowerToysVersion"),
```
since that information is traced already by `EventLaunch`.